### ISSUE TITLE
Best Practices if needing to recreate Kubernetes cluster using this module.

### ISSUE THREAD

**Issue by shermanericts** [terraform-google-modules/terraform-google-kubernetes-engine#1455]
### TL;DR

I may have a situation in the future where I need to recreate the Kubernetes Cluster and subnet ranges using this module.  While I'm using suffixed entropy for new resources (clusters, subnets and the like), I'm having some issues going end-to-end with this.

### Expected behavior

When re-creating the cluster, the new cluster is created first followed by the old cluster being destroyed.

### Observed behavior

When, attempting to do this, I get this Error:

```

│ Error: Error waiting for creating GKE cluster: Invalid value for field PrivateClusterConfig.MasterIpv4CidrBlock: 10.0.0.0/28 conflicts with an existing subnet in one of the peered VPCs.
```

### Terraform Configuration

```hcl
#clusters.tf

locals {
  bastion_members = []
  bastion_name    = format("%s-bastion-%s", local.cluster_name, random_id.bastion_name.hex)
  bastion_sa_name = format("%s-bastion", local.project)
  bastion_zone    = format("%s-%s", var.region, var.bastion_machine_zone)

  node_pools = [
   
    {
      auto_repair                 = true,
      auto_upgrade                = true,
      enable_integrity_monitoring = true
      image_type                  = "COS_CONTAINERD"
      machine_type                = var.node_pool_machine_type,
      max_count                   = 4,
      name                        = "main",
      node_metadata               = "GKE_METADATA",
    },
  ]
}

resource "random_id" "bastion_name" { byte_length = 2 }

resource "null_resource" "gke_cluster_with_bastion" {
  depends_on = [
    module.bastion,
    module.cloud-nat,
    module.enabled_google_apis,
    module.gke,
    module.vpc,
  ]
}

module "bastion" {
  source     = "terraform-google-modules/bastion-host/google"
  depends_on = [random_id.bastion_name, ]

  host_project         = module.enabled_google_apis.project_id
  image_family         = "ubuntu-2204-lts"
  image_project        = "ubuntu-os-cloud"
  machine_type         = var.bastion_machine_type
  members              = local.bastion_members
  name                 = local.bastion_name
  network              = module.vpc.network_self_link
  project              = module.enabled_google_apis.project_id
  service_account_name = local.bastion_sa_name
  shielded_vm          = true
  startup_script       = "sudo apt-get update && sudo apt-get upgrade -y && sudo apt-get install -y tinyproxy"
  subnet               = module.vpc.subnets_self_links[0]
  zone                 = local.bastion_zone
}

module "gke" {
  source     = "github.com/terraform-google-modules/terraform-google-kubernetes-engine//modules/beta-private-cluster-update-variant?ref=v23.3.0"
  depends_on = [module.gcp_project, ]

  cluster_telemetry_type   = "ENABLED"
  enable_private_endpoint  = false
  enable_private_nodes     = true
  grant_registry_access    = true
  ip_range_pods            = module.vpc.subnets_secondary_ranges[0].*.range_name[0]
  ip_range_services        = module.vpc.subnets_secondary_ranges[0].*.range_name[1]
  name                     = local.cluster_name
  network                  = module.vpc.network_name
  network_policy           = true # internally: network_policy=true  ==  network_policy_config.disabled=false
  node_pools               = local.node_pools
  project_id               = module.enabled_google_apis.project_id
  region                   = var.region
  regional                 = true
  release_channel          = "RAPID"
  remove_default_node_pool = true
  subnetwork               = module.vpc.subnets_names[0]
  zones                    = var.zones

  cluster_autoscaling = {
    "autoscaling_profile" : "BALANCED",
    "enabled" : true,
    "gpu_resources" : [],
    "max_cpu_cores" : 24,
    "max_memory_gb" : 96,
    "min_cpu_cores" : 8,
    "min_memory_gb" : 8,
  }

  master_authorized_networks = [{
    cidr_block   = "${module.bastion.ip_address}/32",
    display_name = "Bastion Host",
  }]
}

module "vpc" {
  source = "terraform-google-modules/network/google"

  network_name = local.network_name
  project_id   = module.enabled_google_apis.project_id
  routing_mode = "GLOBAL"

  secondary_ranges = {
    (local.subnet_name) = [
      {
        ip_cidr_range = var.cidr_pods,
        range_name    = "cidr-pods",
      },
      {
        ip_cidr_range = var.cidr_services,
        range_name    = "cidr-services",
      },
    ],
  }

  subnets = [
    {
      description           = "This subnet is managed by Terraform",
      subnet_flow_logs      = true,
      subnet_ip             = var.subnet_ip,
      subnet_name           = local.subnet_name,
      subnet_private_access = true,
      subnet_region         = var.region,
    },
  ]
}

module "cloud-nat" {
  source = "terraform-google-modules/cloud-nat/google"

  create_router = true
  network       = module.vpc.network_self_link
  project_id    = module.enabled_google_apis.project_id
  region        = var.region
  router        = "${local.project}-safer-router"
}

```

# env.tf
```
locals { env_shrm4 = "shrm4" }

resource "random_id" "shrm4_env" { byte_length = 2 }

module "armsctl_shrm4" {
  source = "../terraform-armsctl"

  env     = local.env_shrm4
  project = format(local.project_base, local.env_shrm4)
  region  = local.us-central1.region
  zones   = local.us-central1.zones

  cidr_pods     = "172.0.0.0/16"
  cidr_services = "172.64.0.0/16"
  subnet_ip     = "192.168.0.0/16"

  bastion_machine_type = "g1-small"
  
  cloud_sql_database_flags = { "max_connections" : 400, "cloudsql.iam_authentication" : "on", }
  cloud_sql_machine_type   = "db-custom-1-3840"
  node_pool_machine_type   = "e2-medium"

  customer_projects = []
}
```
```


### Terraform Version

```sh
version
Terraform v1.3.3
on darwin_arm64
+ provider registry.terraform.io/hashicorp/google v4.42.0
+ provider registry.terraform.io/hashicorp/google-beta v4.42.0
+ provider registry.terraform.io/hashicorp/kubernetes v2.15.0
+ provider registry.terraform.io/hashicorp/null v3.2.0
+ provider registry.terraform.io/hashicorp/random v3.4.3
+ provider registry.terraform.io/integrations/github v4.31.0

Your version of Terraform is out of date! The latest version
is 1.3.4. You can update by downloading from https://www.terraform.io/downloads.html
```

### Additional information

Since it's creating before destroying, the old subnets and cluster are still in place - it doesn't appear that we can configure the `PrivateClusterConfig.MasterIpv4CidrBlock` . Any pointers welcome. thanks!

**Comment by shermanericts** [comment]
I wound up figuring out a way to expand this, but it involved doing a targeted destroy of the cluster first.  This usually isn't an event I'd want to do, so if that's the way to do this then we can just close this issue. Thx
